server/app/util/filter.py

```diff
@@ -35,36 +49,47 @@ def parse_filter(filter, schema):
     The results is a matrix with the cells the pass the filter and at this point all the genes
     :param filter: flask's request.args
     :param schema: dictionary schema
+    :raises QueryStringError
     :return:
     """
     query = {}
+
     for key in filter:
```

Comment:
fyi, filter is a python keyword.  may be confusing to use it as a variable name.

Reply:
also, what's the type of filter?  can you extract the key + value at the same time like a dict?

Reply:
It's ImmutableMultiDict type. The reason I'm not getting the values is that I need to use `getlist` method if it is a categorical type and `get` if it is a numerical type.